asignacion_db.sql

```diff
@@ -42,7 +42,7 @@ CREATE TABLE IF NOT EXISTS crm (
             FOREIGN KEY (clave_de_agente_id)
                 REFERENCES agente(clave_de_agente)
                     ON UPDATE CASCADE
-                    ON DELETE RESTRICT
```

Comment:
si cambias el restrict por cascade permites borrar elementos pero cuando borres un registro padre se lleva a todos los hijos lo que evita tener persistencia en la informacion.
por que elegiste cascade ?

Reply:
se cambiaron  por on delete not null, de esta manera se evitará la perdida de los registros hijos si se borra el registro padre.